### ISSUE TITLE
getSymbolsWithAnnotation not working properly with JS/IR

### ISSUE THREAD

**Issue by neetopia** [google/ksp#728]

**Comment by glureau** [comment]
[Sample to reproduce](https://github.com/glureau/ksp-kmp-issues/tree/return_type_not_resolved)

Environment: KSP when used with KotlinJs (IR & Legacy) on a multi-module project.

Expected: resolving a type from another (KMP) module should work

Actual (1.6.0-1.0.1): Resolving fails as soon as the type is defined in an external module

Results of my personal investigation (I made some suppositions, just trying to help):
- The [KtTypeReference.lookup()](https://github.com/google/ksp/blob/main/compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt#L634) extension does return an error type
- ResolverImpl resolves the references thanks to the ModuleDescriptor/BindingTrace
- ModuleDescriptor is configured from the Kotlin Compiler AnalyzerFacade classes
- [TopDownAnalyzerFacadeForJVM](https://github.com/JetBrains/kotlin/blob/master/compiler/cli/src/org/jetbrains/kotlin/cli/jvm/compiler/TopDownAnalyzerFacadeForJVM.kt) compute dependencies but [TopDownAnalyzerFacadeForJS](https://github.com/JetBrains/kotlin/blob/master/js/js.frontend/src/org/jetbrains/kotlin/js/analyze/TopDownAnalyzerFacadeForJS.kt) does not. It only adds `kotlin` and `builtins` in the ModuleDescriptor, so I suppose BindingTrace lookup is limited to that 2 base modules when resolving.

**Comment by SalomonBrys** [comment]
Another reproducer:

- Clone https://github.com/Kodein-Framework/Micro-Mock
- Chek-out the `no-trick` branch
- Run `./gradlew :test:kspTestKotlinJsLegacy` and see that the processor does find annotated symbols:
    ```
    w: [ksp] org.kodein.micromock.Mock: 1
    w: [ksp] org.kodein.micromock.Fake: 1
    w: [ksp] org.kodein.micromock.UsesMocks: 2
    ```
- Run .`/gradlew :test:kspTestKotlinJsIr` and see that the processor does not find any annotated symbols:
    ```
    w: [ksp] org.kodein.micromock.Mock: 0
    w: [ksp] org.kodein.micromock.Fake: 0
    w: [ksp] org.kodein.micromock.UsesMocks: 0
    ```

**Comment by ting-yuan** [comment]
Fixed in #761
